scala/private/rule_impls.bzl

```diff
@@ -82,6 +83,8 @@ def compile_scala(
     args.add("--StatsfileOutput", statsfile)
     args.add("--EnableDiagnosticsReport", enable_diagnostics_report)
     args.add("--DiagnosticsFile", diagnosticsfile)
+    args.add("--Release", java_toolchain.target_version)
```

Comment:
would this be able to accommodate using jdk 11 to compile scala targeting 8 (so `-release 8`), then run on 11?

Reply:
This is still a wip. I don't think `-release` flag is a solution (especially this is not available in scala 2.11). I'm trying `javabootclasspath` now.

However I think you could add `scalacopts = ["-release", "8"]` to your `scala_toolchain` to get unblocked (referring to https://github.com/bazelbuild/rules_scala/issues/1233)

Reply:
If 
> add scalacopts = ["-release", "8"]

and 
```
    args.add("--Release", java_toolchain.target_version)
```

Are they the same thing, or does one take precedence over the other?

Reply:
`args.add("--Release", java_toolchain.target_version)` translates to `"-release", "x"` [here](https://github.com/bazelbuild/rules_scala/pull/1236/files#diff-ee9bbb81b4acc0673a726250f8fee1a76e466b308c5e29148fdac091d3f37657R246)

Of course if we implicitly add `-release` it might conflict with user supplied flag (and we need to handle that).
`-release 8` passes test case `bazel run --javabase=:jdk8 --host_javabase=:jdk11 :MainScala` but is not enough for all jdk and scala versions.

Reply:
I see. Thanks for clarifying! that works.